### PR TITLE
Converted all pages to React + other imp changes

### PR DESCRIPTION
**Pull request by Devansu-Yadav**
Added UI to React for the following pages:
- Landing Page
- Product listing page and Filters
- WishList page
- Cart Management page
- Auth pages(Login/Signup).

Other important changes:-
- Setup the Folder Structure for the React app.
- Added routes for all the pages.
- Added data for products, wishList, and cart pages.

### CODE REVIEW

File: src/components/Cart/CartList.jsx [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+        setCartListItemsCount(initialCartListItemsCount);
+    }, []);
+
+    // Show Empty CartList if items count is zero
+    useEffect(() => {
+        !cartListItemsCount ? setShowEmptyCart(true): setShowEmptyCart(false);
+    }, [cartListItemsCount]);
+
+    // User Cart Data will be fetched from the API.
+    const cartItems = [
```
Comment (shrey27) [inline src/components/Cart/CartList.jsx:23]: cart items should be fetched from db

File: src/components/Cart/CartList.jsx [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+
+                {/* Cart Cards  */}
+                <div className="cart">
+                    <div className={`cart-cards ${showEmptyCart ? "empty": ""} cart-grid-2-column`}>
+                        <CartListCard productData={{
+                            image: cartItems[0].image,
+                            productName: cartItems[0].productName,
+                            price: cartItems[0].price
+                        }} cartListItemsCountHandler={setCartListItemsCount}/>
+
```
Comment (shrey27) [inline src/components/Cart/CartList.jsx]: use map instead of hardcoding each CartListCard

**Review by shrey27 — APPROVED** [review]
overall code looks up to the mark
file structure is also good
I have suggested few changes, but apart from that PR is good to merge

File: src/pages/LoginPage.jsx [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+import { useSidebar } from "../common/context/SidebarContext";
+import "../styles/LoginPage.css";
+
+const LoginPage = () => {
+    const { displaySideBar } = useSidebar();
+
+    // Updating title on rendering Login Page comp
+    useEffect(() => {
+        document.title = "BlueKart - Login";
+    }, []);
```
Comment (shrey27) [inline src/pages/LoginPage.jsx:15]: that's a nice way of using page name as title

File: src/components/Cart/CartList.jsx (hunk 1 above)
Comment (Devansu-Yadav) [reply src/components/Cart/CartList.jsx:23]: Yeah, once I integrate the API for fetching cart items, there will be no need of hardcoding it 😅

File: src/components/Cart/CartList.jsx (hunk 2 above)
Comment (Devansu-Yadav) [reply src/components/Cart/CartList.jsx]: Thanks, Shrey, actually I didn't realize I could use `map` here ... will make this change asap😅

File: src/common/helpers/priceFormatter.js [hunk 4]
```diff
@@ -0,0 +1,6 @@
+// Useful for Formatting the price acc to Indian pricing system to be displayed on Product Cards.
+const formatProductPrice = (price) => {
```
Comment (romabulani) [inline src/common/helpers/priceFormatter.js:2]: Nice use of comments

**Review by romabulani — APPROVED** [review]
As Shrey already mentioned, you should remove the cart and wishlist in further PRs
- Nice comments and folder structure
- Good to merge

**Pull request merged**